[2025-01-13T21:15:32.873+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: finflow_source_data_pipeline.data_quality_checks.verify_merge_results_date manual__2025-01-13T21:13:17.052421+00:00 [queued]>
[2025-01-13T21:15:33.097+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: finflow_source_data_pipeline.data_quality_checks.verify_merge_results_date manual__2025-01-13T21:13:17.052421+00:00 [queued]>
[2025-01-13T21:15:33.113+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2025-01-13T21:15:33.274+0000] {taskinstance.py:1382} INFO - Executing <Task(BigQueryExecuteQueryOperator): data_quality_checks.verify_merge_results_date> on 2025-01-13 21:13:17.052421+00:00
[2025-01-13T21:15:33.369+0000] {standard_task_runner.py:57} INFO - Started process 19259 to run task
[2025-01-13T21:15:33.389+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'finflow_source_data_pipeline', 'data_quality_checks.verify_merge_results_date', 'manual__2025-01-13T21:13:17.052421+00:00', '--job-id', '1153', '--raw', '--subdir', 'DAGS_FOLDER/finflow.py', '--cfg-path', '/tmp/tmpb9u1d6tr']
[2025-01-13T21:15:33.445+0000] {standard_task_runner.py:85} INFO - Job 1153: Subtask data_quality_checks.verify_merge_results_date
[2025-01-13T21:15:34.012+0000] {task_command.py:416} INFO - Running <TaskInstance: finflow_source_data_pipeline.data_quality_checks.verify_merge_results_date manual__2025-01-13T21:13:17.052421+00:00 [running]> on host ab4e86dc450e
[2025-01-13T21:15:35.142+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='finflow_source_data_pipeline' AIRFLOW_CTX_TASK_ID='data_quality_checks.verify_merge_results_date' AIRFLOW_CTX_EXECUTION_DATE='2025-01-13T21:13:17.052421+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-01-13T21:13:17.052421+00:00'
[2025-01-13T21:15:35.148+0000] {bigquery.py:1241} INFO - Executing: 
WITH pre_counts AS (
    SELECT 
        'dates' as table_name,
        COUNT(*) as pre_merge_count
    FROM `finflow-analytics-production.finflow_data.src_dates`
    WHERE DATE(ingestion_timestamp) < DATE('2025-01-13')
),
post_counts AS (
    SELECT 
        'dates' as table_name,
        COUNT(*) as post_merge_count
    FROM `finflow-analytics-production.finflow_data.src_dates`
    WHERE DATE(ingestion_timestamp) <= DATE('2025-01-13')
),
new_records AS (
    SELECT 
        'dates' as table_name,
        COUNT(*) as new_record_count
    FROM `finflow-analytics-production.finflow_data.src_dates`
    WHERE DATE(ingestion_timestamp) = DATE('2025-01-13')
)
SELECT
    CASE 
        WHEN post_counts.post_merge_count < pre_counts.pre_merge_count 
        THEN ERROR('Data loss detected in dates table')
        WHEN post_counts.post_merge_count - pre_counts.pre_merge_count != new_records.new_record_count
        THEN ERROR('Unexpected record count after merge in dates table')
        ELSE 'Merge validation successful for dates table'
    END as validation_result
FROM pre_counts
JOIN post_counts ON pre_counts.table_name = post_counts.table_name
JOIN new_records ON pre_counts.table_name = new_records.table_name
[2025-01-13T21:15:35.167+0000] {crypto.py:82} WARNING - empty cryptography key - values will not be stored encrypted.
[2025-01-13T21:15:35.180+0000] {connection.py:232} WARNING - Connection schemes (type: google_cloud_platform) shall not contain '_' according to RFC3986.
[2025-01-13T21:15:35.194+0000] {base.py:73} INFO - Using connection ID 'google_cloud_default' for task execution.
[2025-01-13T21:15:35.230+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/providers/google/cloud/hooks/bigquery.py:2108: AirflowProviderDeprecationWarning: This method is deprecated. Please use `BigQueryHook.insert_job` method.
  warnings.warn(

[2025-01-13T21:15:36.041+0000] {bigquery.py:1595} INFO - Inserting job ***_1736802936039633_31135aff66f877ba9b073acdff1b53e7
[2025-01-13T21:15:37.217+0000] {taskinstance.py:1400} INFO - Marking task as SUCCESS. dag_id=finflow_source_data_pipeline, task_id=data_quality_checks.verify_merge_results_date, execution_date=20250113T211317, start_date=20250113T211532, end_date=20250113T211537
[2025-01-13T21:15:37.356+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2025-01-13T21:15:37.566+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
